src/game/Creature.cpp

```diff
@@ -2247,6 +2247,27 @@ bool Creature::MeetsSelectAttackingRequirement(Unit* pTarget, SpellEntry const*
         if (selectFlags & SELECT_FLAG_PLAYER && pTarget->GetTypeId() != TYPEID_PLAYER)
             return false;
 
+        if (selectFlags & SELECT_FLAG_PLAYER_CLASS_WARRIOR && (pTarget->GetTypeId() != TYPEID_PLAYER || pTarget->getClass() != CLASS_WARRIOR))
```

Comment:
switch case does not work in this case.

If we use a switch case, then you've moved the getclass to the beginning of the functionality, while it needs to be in the end, otherwise we may as well just have done a switch case on the power thing as well.

The reason why I do it as else if and not just a bunch of if's is because you are not able to send along 2 player selectoin flags, just as you are not able to send along 2 power selection flags, as with the current logic it would return false if it found the wrong class on the first, not even looking at the second flag - to solve this the flags would have to be reversed to the inverse, a NOT WARRIOR flag, and a NOT MANA flag, that way you're not excluding based on what people aren't, but rather on what they have.

I will repeat; this is the same method as used for finding power, if you have any thought out solutions please suggest them.